### PR TITLE
ksm_overcommit: Fix some issues

### PR DESCRIPTION
**Pull request by huangyum**
1. Remove pid file from qemu cli
2. Fix if condition about 32 bit guests

Signed-off-by: Yumei Huang <yuhuang@redhat.com>
ID: 1876448

### CODE REVIEW

**Comment by huangyum** [comment]
@yanan-fu Would you please help review? 
The case may run for quite a  few hours.  I modified the guest memory to a smaller value temporarily and ran the case, test passed.
 (1/2) Host_RHEL.m8.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.ksm_overcommit.ksm_serial: PASS (514.06 s)
 (2/2) Host_RHEL.m8.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.ksm_overcommit.ksm_parallel: PASS (188.34 s)

File: qemu/tests/ksm_overcommit.py [hunk 1]
```diff
@@ -506,9 +508,13 @@ def split_parallel():
         # mem: Memory of the guest systems. Maximum must be less than
         # host's physical ram
         mem = int(overcommit * host_mem / vmsc)
+        # If guest memory is too much, it will take very very long time for ksm
+        # to scan and merge same pages, sometimes will hit timeout
+        if mem > 8192:
+            mem = 8192
```
Comment (yanan-fu) [inline qemu/tests/ksm_overcommit.py:514]: With this change, no overcommit if host mem is large, but just kvm function be tested.  I wonder if it is enough for the test scenario.

File: qemu/tests/ksm_overcommit.py [hunk 2]
```diff
@@ -449,7 +450,7 @@ def split_parallel():
     if (guest_reserve == -1):
         # default guest_reserve = minimal_system_mem(256MB)
         # later we add tmpfs overhead
-        guest_reserve = 256
+        guest_reserve = 1536
```
Comment (yanan-fu) [inline qemu/tests/ksm_overcommit.py:453]: Could you explain more about this ?  And the comment in L451 need be updated accordingly.

File: qemu/tests/ksm_overcommit.py [hunk 3]
```diff
@@ -151,7 +152,7 @@ def initialize_guests():
                           ksm_size)
             while ((new_ksm and (shm < (ksm_size * (i + 1)))) or
                     (not new_ksm and (shm < (ksm_size)))):
-                if j > 64:
+                if j > 256:
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/ksm_overcommit.py:155]: It's just to add more times to check pages.  Will add the detailed info to commit message. Thanks.

File: qemu/tests/ksm_overcommit.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/ksm_overcommit.py:514]: Yes, when host memory is large, there might not be overcommit happens. But still ksm should work. For large host, ksm would take much much more time to scan and merge pages, even 2 days are not long enough in my test. So I just readjust guest memory. I think the ksm function is the most important thing we care.

File: qemu/tests/ksm_overcommit.py (hunk 2 above)
Comment (huangyum) [reply qemu/tests/ksm_overcommit.py:453]: 256M is not enough now. We may hit OOM easily, which is not what we desire.  As far as I'm concerned, 1.5G is more safer.

**Comment by huangyum** [comment]
@yanan-fu Would you pls help review again, thanks.

**Comment by fbq815** [comment]
Test result:
JOB ID     : 49e77f21a5eb6cc8e0661e7219f26c75570a42de
JOB LOG    : /root/avocado/job-results/job-2021-12-20T00.44-49e77f2/job.log
 (1/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (372.61 s)
 (2/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.ksm_overcommit.ksm_serial.s390-virtio: PASS (1654.81 s)
 (3/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.ksm_overcommit.ksm_parallel.s390-virtio: PASS (53.30 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2021-12-20T00.44-49e77f2/results.html
JOB TIME   : 2082.51 s

**Review by fbq815 — APPROVED** [review]

**Comment by huangyum** [comment]
@yanan-fu Any more comments?

File: qemu/tests/ksm_overcommit.py [hunk 4]
```diff
@@ -631,11 +615,6 @@ def split_parallel():
                        "living VM" % lvms[i].name)
 
         lsessions.append(lvms[i].wait_for_login(timeout=360))
-        try:
-            tmp = open(params.get('pid_' + vm_name), 'r')
-            params['pid_' + vm_name] = int(tmp.readline())
-        except Exception:
-            test.fail("Could not get PID of %s" % (vm_name))
```
Comment (yanan-fu) [inline qemu/tests/ksm_overcommit.py:618]: Checking if QEMU process PID be stored the pid file is not the check point for this case ?

Comment (yanan-fu) [reply qemu/tests/ksm_overcommit.py:618]: In fact, i am ok with the change that remove pidfile,  i do not think this file makes mush sense.  But a check point be removed with this change.  If no meaning for ksm test, i am ok.

Comment (huangyum) [reply qemu/tests/ksm_overcommit.py:618]: It has nothing to do with ksm feature. So no need to check that. Thanks.

**Review by yanan-fu — APPROVED** [review]
Ack

**Pull request merged**